tests/stubs/talon/__init__.py

```diff
@@ -16,30 +25,78 @@ class Actions:
     """
 
     def __init__(self):
-        self.reset_actions()
+        self.registered_actions = {
+            "module": {"non_namespaced": {}, "namespaced": {}},
+            "test": {"non_namespaced": {}, "namespaced": {}},
+            # "contexts": []
+        }
+
+        # Some built in actions
+        self.register_module_action("", "key", lambda x: None)
+        self.register_module_action("", "insert", lambda x: None)
+        self.register_module_action("", "sleep", lambda x: None)
+        self.register_module_action("edit", "selected_text", lambda: "test")
+
+    def reset_test_actions(self):
+        self.registered_actions["test"] = {"non_namespaced": {}, "namespaced": {}}
+
+    def register_module_action(self, namespace: str, name: str, func: Callable):
+        """
+        Registers an action to the module category. This should
+        only be called by importing files containing module definitions.
+        It won't be reset between test runs (or test files). Use
+        register_test_action and reset_test_actions to temporarily override
+        actions.
+        """
 
-    def reset_actions(self):
-        self.registered_actions = {"user": {}, "edit": {}}
-        self.edit = self._build_namespace_accessor("edit")
-        self.user = self._build_namespace_accessor("user")
+        self._register_action("module", namespace, name, func)
 
-    def register_test_action(self, name: str, func: Callable):
+    def register_test_action(self, namespace: str, name: str, func: Callable):
         """
         Registers the given action, use like:
 
             actions.register("user.my_action", lambda: None)
         """
 
-        namespace, action_name = name.split(".")
-        self.registered_actions[namespace][action_name] = func
+        self._register_action("test", namespace, name, func)
 
-    def key(self, key):
-        """
-        Stub out actions.key
-        """
+    def _register_action(
+        self, category: str, namespace: str, name: str, func: Callable
+    ):
+        if namespace == "":
+            self.registered_actions[category]["non_namespaced"][name] = func
+        else:
+            if namespace not in self.registered_actions[category]["namespaced"]:
+                self.registered_actions[category]["namespaced"][namespace] = {}
+
+            self.registered_actions[category]["namespaced"][namespace][name] = func
+
+    def __getattr__(self, name):
+        callable = None
+        try:
+            callable = object.__getattribute__(self, name)
+        except AttributeError:
+            pass
+
+        if callable:
+            return callable
 
-    def _build_namespace_accessor(self, key):
-        return DictAccessor(self.registered_actions[key])
+        found_namespace = False
+        for category in ("test", "module"):
```

Comment:
This duplicates some of the logic in RegisteredActionsAccessor. Would be cleaner to simply have a RegisteredActionsAccessor for "main" and try using it first, then falling back to RegisteredActionsAccessor(self.registered_actions, name) and letting that constructor throw an error if the namespace doesn't exist instead of doing the RuntimeError here.